File: Backend/email/src/main/java/com/example/email/model/Email.java
package com.example.email.model;

import java.util.Date;

import org.springframework.data.annotation.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Email {

    @Id
    private String id;

    private String from;

    private String to;

    private String subject;

    private String body;

    private Date date;

    private Boolean read;

    private Boolean draft;
}

File: Frontend/email/src/app/models/email.ts
import { Attachment } from "./attachment";

export interface Email {
  id: string;
  from: string;
  to: string;
  subject: string;
  body: string;
  date: string;
  read: boolean;
  draft: boolean;
  attachments?: Attachment[];
  attachmentsId?: string[];
}
